pkg/svcat/service-catalog/instance.go

```diff
@@ -160,6 +160,10 @@ func (sdk *SDK) Provision(namespace, instanceName, className, planName string,
 		},
 	}
 
+	if externalID != "" {
```

Comment:
Same as before, this can be unconditionally set when the request is initialized.